src/lib.rs

```diff
@@ -79,14 +87,33 @@ pub fn get_args() -> DCPResult<Config> {
     })
 }
 
+// Returns a Result with an Image struct and an Error (if one occured)
+fn split_repo_and_tag(full_image: String) -> Image {
```

Comment:
could we make function part of the Image struct? Using `Impl Image` we define the types methods, and we can add this one there.